### ISSUE TITLE
[1.1.4-20190826] java.lang.NoSuchMethodError: com.sun.jna.Native.load(Ljava/lang/String;Ljava/lang/Class;Ljava/util/Map;)Lcom/sun/jna/Library;

### ISSUE THREAD

**Issue by kriazhev** [RaiMan/SikuliX1#190]
Updated to the **1.1.4-20190826** from 1.1.4-20181214

`java.lang.NoSuchMethodError: com.sun.jna.Native.load(Ljava/lang/String;Ljava/lang/Class;Ljava/util/Map;)Lcom/sun/jna/Library;
	at org.sikuli.natives.SXUser32.<clinit>(SXUser32.java:13)
	at org.sikuli.natives.WinUtil.<clinit>(WinUtil.java:37)
	at org.sikuli.script.support.RunTime.addToWindowsSystemPath(RunTime.java:1664)
	at org.sikuli.script.support.RunTime.libsExport(RunTime.java:1607)
	at org.sikuli.script.support.RunTime.libsLoad(RunTime.java:1432)
	at org.sikuli.script.support.RunTime.loadLibrary(RunTime.java:1636)
	at org.sikuli.script.Finder$Finder2.<clinit>(Finder.java:571)
	at org.sikuli.script.Finder$FindInput2.<clinit>(Finder.java:1210)
	at org.sikuli.script.Finder.<init>(Finder.java:36)
	at org.sikuli.script.Region.doCheckLastSeenAndCreateFinder(Region.java:2951)
	at org.sikuli.script.Region.checkLastSeenAndCreateFinder(Region.java:2912)
	at org.sikuli.script.Region.doFind(Region.java:2839)
	at org.sikuli.script.Region.access$300(Region.java:23)
	at org.sikuli.script.Region$RepeatableFind.run(Region.java:3122)
	at org.sikuli.script.Region$Repeatable.repeat(Region.java:3073)
	at org.sikuli.script.Region.wait(Region.java:2258)
`

I'm on Gradle
sikulixapi transitively downloads
 net.java.dev.jna:jna:4.5.2
 net.java.dev.jna:jna-platform:4.5.2

Tried to specify newer versions(5.3.2, 5.0.0, 5.4.0) of JNA that ended with another exception
` java.lang.ClassNotFoundException: com.sun.jna.platform.win32.WinUser$WNDENUMPROC
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.sikuli.script.support.RunTime.addToWindowsSystemPath(RunTime.java:1664)
	at org.sikuli.script.support.RunTime.libsExport(RunTime.java:1607)
	at org.sikuli.script.support.RunTime.libsLoad(RunTime.java:1432)
	at org.sikuli.script.support.RunTime.loadLibrary(RunTime.java:1636)
	at org.sikuli.script.Finder$Finder2.<clinit>(Finder.java:571)
	at org.sikuli.script.Finder$FindInput2.<clinit>(Finder.java:1210)
	at org.sikuli.script.Finder.<init>(Finder.java:36)
	at org.sikuli.script.Region.doCheckLastSeenAndCreateFinder(Region.java:2951)
	at org.sikuli.script.Region.checkLastSeenAndCreateFinder(Region.java:2912)
	at org.sikuli.script.Region.doFind(Region.java:2839)
	at org.sikuli.script.Region.access$300(Region.java:23)
	at org.sikuli.script.Region$RepeatableFind.run(Region.java:3122)
	at org.sikuli.script.Region$Repeatable.repeat(Region.java:3073)
	at org.sikuli.script.Region.wait(Region.java:2258)`

**Comment by balmma** [comment]
1.1.4 currently uses JNA 5.3.1
```
<dependency>
      <groupId>net.java.dev.jna</groupId>
      <artifactId>jna-platform</artifactId>
      <version>5.3.1</version>
</dependency>
```
But I guess it should work with 5.3.2 as well.

**Comment by balmma** [comment]
Just FYI: The JNA dependency got updated with PR #127.

**Comment by kriazhev** [comment]
Thank you for prompt reply.

I think my question is a bit cumbersome. Let me try again.

1. I have gradle project
2. I've downloaded https://oss.sonatype.org/content/repositories/snapshots/com/sikulix/sikulixapi/1.1.4-SNAPSHOT/sikulixapi-1.1.4-20190905.080752-255.jar
3. Added the dependency to my gradle project: implementation files("C:\\Data\\sikulixapi-1.1.4-20190905.080752-255.jar")
4. Run gradle test task
5. Got an exception 
`com/sun/jna/platform/win32/WinUser$WNDENUMPROC
java.lang.NoClassDefFoundError: com/sun/jna/platform/win32/WinUser$WNDENUMPROC
	at org.sikuli.script.support.RunTime.addToWindowsSystemPath(RunTime.java:1663)`

6. Added jna-platform dependency to my gradle project: implementation "net.java.dev.jna:jna-platform:5.3.1"
7. Got another exception
`com.sun.jna.Native.load(Ljava/lang/String;Ljava/lang/Class;Ljava/util/Map;)Lcom/sun/jna/Library;
java.lang.NoSuchMethodError: com.sun.jna.Native.load(Ljava/lang/String;Ljava/lang/Class;Ljava/util/Map;)Lcom/sun/jna/Library;
	at org.sikuli.natives.SXUser32.<clinit>(SXUser32.java:13)`


Could you please advice on my next steps?

**Comment by balmma** [comment]
Are you sure that there is no other JNA version lingering around in your classpath?

**Comment by balmma** [comment]
PR #191 removes the IMHO unnecessary enumWindows call and might mitigate your issue here. But I'm still quite sure that you have a wrong version of JNA in your classpath.
